pkg/drivers/powerflex.go

```diff
@@ -371,28 +379,41 @@ func ExtractZonesFromSecret(ctx context.Context, kube client.Client, namespace s
 		yamlConfig := make([]StorageArrayConfig, 0)
 		configs, err := yaml.JSONToYAML(configBytes)
 		if err != nil {
-			return nil, fmt.Errorf("unable to parse multi-array configuration[%v]", err)
+			return fmt.Errorf("unable to parse multi-array configuration %v", err)
 		}
 		err = yaml.Unmarshal(configs, &yamlConfig)
 		if err != nil {
-			return nil, fmt.Errorf("unable to unmarshal multi-array configuration[%v]", err)
+			return fmt.Errorf("unable to unmarshal multi-array configuration %v", err)
 		}
 
+		var labelKey string
+		var numArrays, numArraysWithZone int
+		numArrays = len(yamlConfig)
 		for _, configParam := range yamlConfig {
 			if configParam.SystemID == "" {
-				return nil, fmt.Errorf("invalid value for SystemID")
+				return fmt.Errorf("invalid value for SystemID")
+			}
+			if labelKey == "" {
+				labelKey = configParam.Zone.LabelKey
+			} else {
+				if labelKey != configParam.Zone.LabelKey {
+					return fmt.Errorf("labelKey not consistent across all arrays")
+				}
 			}
 			if configParam.Zone.Name != "" {
+				numArraysWithZone++
 				zonesMapData[configParam.SystemID] = configParam.Zone.Name
 				log.Infof("Zoning information configured for systemID %s: %v ", configParam.SystemID, zonesMapData)
-			} else {
-				log.Info("Zoning information not found in the array config. Continue with topology-unaware driver installation mode")
-				return zonesMapData, nil
 			}
 		}
+		if numArraysWithZone > 0 && numArrays != numArraysWithZone {
+			return fmt.Errorf("not all arrays have zoning configured. Check the array info secret, zone key should be the same for all arrays")
+		} else if numArraysWithZone == 0 {
+			log.Info("Zoning information not found in the array config. Continue with topology-unaware driver installation mode")
+		}
```

Comment:
This validation will give a false negative when first array(s) has no zoning, but the following array(s) do have some zoning configuration. A unit test should be added for this scenario.

Also, I'd suggest this greatly simplified version, e.g.
```
		var labelKey string
		for i, configParam := range yamlConfig {
			// remember the labelKey from the first array, and make sure all arrays have the same labelKey
			if i == 0 {
				labelKey = configParam.Zone.LabelKey
			} else if labelKey != configParam.Zone.LabelKey {
				return fmt.Errorf("labelKey not consistent across all arrays in array secret")
			}
			// make sure zone Name is not empty when labelKey is present
			if configParam.Zone.LabelKey != "" && configParam.Zone.Name == "" {
				return fmt.Errorf("zone name is empty for some arrays in array secret")
			}
		}
```

Reply:
ok, there's already a unit test for this. the code will handle this scenario just fine.

Reply:
There are specific scenarios that are not handled by this code, namely partially specified zone structure, zone.LabelKey or zone.Name This is addressed in the latest changes.